src/mesh/mod.rs

```diff
@@ -213,35 +226,72 @@ impl ShapeMesh {
             }
         }
 
-
         Ok(())
     }
 
-    /// Returns the point on the shape's surface the camera is currently looking at
-    pub fn get_focus(&self, camera: &Camera) -> Option<Point3<f32>> {
+    /// Returns equally distributed points on the near_plane. The number of points
+    /// returned is focus_points².
+    pub fn get_focii(&self, camera: &Camera, focus_points: u8) -> Vec<Point3<f32>> {
+        // sets the amount of focus points that may cause a split event
         const EPSILON: f32 = 0.000_001;
         const MAX_ITERS: u64 = 100;
 
-        let mut pos = camera.position;
-        for _ in 0..MAX_ITERS {
-            let distance = self.shape.min_distance_from(pos);
-            pos += camera.direction() * distance;
-            if distance < EPSILON {
-                break;
+        let bb = camera.get_near_plane_bb();
+        let frustum_width = bb.1.x - bb.0.x;
+        let frustum_height = bb.1.y - bb.0.y;
+        let size_horizontal = frustum_width/focus_points as f32;
+        let size_vertical = frustum_height/focus_points as f32;
+        let center_diff = (bb.1 - bb.0)/(2.0*focus_points as f32);
+
+        let inv_view_trans = camera.view_transform().invert().unwrap();
+
+        let mut vec = Vec::new();
```

Comment:
For the rest of this method we can make use of awesome Rust iterators. 

First, we can use the `util::iter::cube()` method here to return an iterator over ... x and y coordinates of a cube. So without using iterators yet, you could replace both of your `for`-loops with 

```rust
use util::iter; // somewhere at the top of the file

for (x, y) in iter::cube(focus_points) { ... }
```

Awesome, right? But instead of pushing all immediate results to the vector yet, we can just keep the iterator and apply the next step onto it. Here is some code with some explanation in comments:

```rust
iter::cube(focus_points)  // at this point we iterate over the x/y pair
    .map(|(x, y)| {
        // here we will "map" the iterator's elements, so for each element, 
        // we will create a new one
        let center = ...;

        // We just return the created point
        Point3::from_homogeneous(
            inv_view_trans * center.to_homogeneous()
        )
    })
    // <-- at this point we no longer iterate over the x/y pairs, but over
    //     the Point3s we created in the `map` operation

    // next we use another iterator method to filter the elements 
    // (discarding some, keeping others)
    .filter(|p| { // note: `p` is the point from the previous step
        // here we have to return `true` to keep the current element or 
        // `false` to discard it

        let mut pos = camera.position;
        let dir = (point - camera.position).normalize();

        for _ in 0..MAX_ITERS {
            let distance = self.shape.min_distance_from(pos);
            pos += dir * distance;
            if distance < EPSILON {
                // We approached the point close enough, so we want
                // to keep it
                return true;
            }
        }

        // We tried for `MAX_ITERS`, but didn't hit the point, so we want to
        // discard it
        false
    })
    // <-- at this point we are still iterating over the points we created
    //     in the `map()` operation, but not necessarily all of them! Our
    //     iterator was possibly shortened in the `filter()` step

    // Now we have an iterator over all the elements we want. We just have to
    // push them all onto a vector. But that's easy too: there is a `collect()`
    // method for iterators, which (in our case) directly returns the vector.
    .collect()
```

I hope the explanation was understandable. Iterators are a truly magic way to write shorter and more descriptive code. If anything is unclear, just ask (possibly via Telegram ^_^).